File: 02/019/main.js
function numberOfSundays() {
    let sundays = 0
    for (let year = 1901; year <= 2000; year++) {
        for (let month = 0; month < 12; month++) {
            const theDay = new Date(year, month, 1)
            if (theDay.getDay() === 0) {
                sundays++
            }
        }
    }
    return sundays
}

console.log(numberOfSundays())

File: 02/019/main.hs
import Data.Time.Format
import Data.Time

dateRanges = [parseDate $ show month ++ " " ++ show year | year <- [1901..2000], month <- [1..12]]
    where
        parseDate strDate = parseTimeOrError True defaultTimeLocale "%-m %Y" strDate :: UTCTime


numberOfSundays = length [date | date <- dateRanges, formatTime defaultTimeLocale "%A" date == "Sunday"] 

main = do
    print numberOfSundays
